src/java.base/share/classes/java/lang/Class.java

```diff
@@ -392,9 +392,10 @@ static String typeVarBounds(TypeVariable<?> typeVar) {
      * caller frame on the stack (e.g. when called directly from a JNI
      * attached thread), the system class loader is used.
      *
-     * @param      className   the fully qualified name of the desired class.
-     * @return     the {@code Class} object for the class with the
-     *             specified name.
+     * @param     className the <a href="#binary-name">binary name</a> of the class
```

Comment:
Where do these links lead? I don't find any section with this ID in the `Class` Javadoc. Should they lead to `ClassLoader`? That Javadoc has such a section.

Reply:
This seems a specification change as the binary-name and FQN are not always the same. In particular for a member class `M` in class `C` the binary name is `C$M`, whereas the FQN is `C.M`.

Reply:
> Binary name is a long-standing behavior. It's a spec bug.

Yes, I view this PR as just specifying long standing behavior. The name used for array class may be surprising but I don't think it can be changed.